src/saga/adaptors/lsf_summit/lsfjob_summit.py

```diff
@@ -0,0 +1,1068 @@
+
+    if jd.error is not None:
+        # if working directory is set, we want stderr to end up in 
+        # the working directory as well, unless it contains a specific
+        # path name. 
+        if jd.working_directory is not None:
+            if os.path.isabs(jd.error):
+                lsf_params += "#BSUB -e %s \n" % jd.error
+            else:
+                # user provided a relative path for STDERR. in this case
+                # we prepend the working directory path before passing
+                # it on to LSF.
+                lsf_params += "#BSUB -e %s/%s \n" % (jd.working_directory, jd.error)
+        else:
+            lsf_params += "#BSUB -e %s \n" % jd.error
+
+
+    if jd.wall_time_limit is not None:
+        hours = jd.wall_time_limit / 60
+        minutes = jd.wall_time_limit % 60
+        lsf_params += "#BSUB -W %s:%s \n" \
+            % (str(hours), str(minutes))
+
+    if (jd.queue is not None) and (queue is not None):
+        lsf_params += "#BSUB -q %s \n" % queue
+    elif (jd.queue is not None) and (queue is None):
+        lsf_params += "#BSUB -q %s \n" % jd.queue
+    elif (jd.queue is None) and (queue is not None):
+        lsf_params += "#BSUB -q %s \n" % queue
+
+    if jd.project is not None and ':' in jd.project:
+        account, reservation = jd.project.split(':',1)
+        lsf_params += "#BSUB -P %s \n" % str(account)
+        lsf_params += "#BSUB -U %s \n" % str(reservation)
+    elif jd.project is not None:
+        lsf_params += "#BSUB -P %s \n" % str(jd.project)
+
+    if jd.job_contact is not None:
+        lsf_params += "#BSUB -u %s \n" % str(jd.job_contact)
+
+    # Request enough nodes to cater for the number of cores requested
+    if jd.total_cpu_count is None:
+        total_cpu_count = 1
+    else:
+        total_cpu_count = jd.total_cpu_count
+
+    ppn = 42
+    number_of_nodes = int(total_cpu_count / ppn)
+    if total_cpu_count % ppn > 0:
+        number_of_nodes += 1
+    lsf_params += "#BSUB -nnodes %s \n" %str(number_of_nodes)
+
+    lsf_params += "#BSUB -alloc_flags 'gpumps smt4' \n"
+
+    # span parameter allows us to influence core spread over nodes
+    if span:
+        lsf_params += '#BSUB -R "span[%s]"\n' % span
+
+    # escape all double quotes and dollar signs, otherwise 'echo |'
+    # further down won't work
+    # only escape '$' in args and exe. not in the params
+    #exec_n_args = workdir_directives exec_n_args
+    exec_n_args = exec_n_args.replace('$', '\\$')
+
+    lsfscript = "\n#!/bin/bash \n%s\n%s\n%s" % (lsf_params, env_variable_list, exec_n_args)
+
+    lsfscript = lsfscript.replace('"', '\\"')
+    return lsfscript
+
+
+# --------------------------------------------------------------------
+# some private defs
+#
+_PTY_TIMEOUT = 2.0
+
+# --------------------------------------------------------------------
+# the adaptor name
+#
+_ADAPTOR_NAME          = "saga.adaptor.lsfjob_summit"
+_ADAPTOR_SCHEMAS       = ["lsf", "lsf+ssh", "lsf+gsissh"]
```

Comment:
This will conflict with the other LSF adaptor - adaptor selection is based on URLs.

Reply:
Yes, you are right.